packages/astro/src/astro-components/control/Protect.astro

```diff
@@ -0,0 +1,38 @@
+---
+import {
+  CheckAuthorizationWithCustomPermissions,
+  OrganizationCustomPermissionKey,
+  OrganizationCustomRoleKey,
+} from "@clerk/types";
+
+type Props =
+  | {
+      condition?: never;
+      role: OrganizationCustomRoleKey;
+      permission?: never;
+    }
+  | {
+      condition?: never;
+      role?: never;
+      permission: OrganizationCustomPermissionKey;
+    }
+  | {
+      condition: (has: CheckAuthorizationWithCustomPermissions) => boolean;
+      role?: never;
+      permission?: never;
+    }
+  | {
+      condition?: never;
+      role?: never;
+      permission?: never;
+    };
```

Comment:
These are common with the types for `Protect` from our Next.js SDK right? Maybe an opportunity to consolidate 👀

Reply:
I see this is declared as `ProtectComponentDefaultProps` elsewhere, can we reuse that?

Reply:
So the idea was, to do minimal changes on the first PR and start removing or cleaning up, afterwards. There is already a ticket for handling/improving sharing logic. I believe what I should do is do another pass identify potential "shared" code, mark it, and document it in the ticket. Once we have e2e tests or a few example apps using the official package then we can safely start improving the shared logic.

Reply:
In this case i think i will just use the already existed type as you recommended.